mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,81 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
+    return res.status(403).jssosn({
+      error: 'Access denied',
+    });
+  }
+  next();
+};
+
+exports.forgotPassword = (req, res) => {
+  if (!req.body) return res.status(400).json({ message: 'No request body' });
+  if (!req.body.email)
+    return res.status(400).json({ message: 'No Email in request body' });
+
+  console.log('forgot password finding user with that email');
+  const { email } = req.body;
+  console.log('signin req.body', email);
+
+  User.findOne({ email }, (err, user) => {
+    if (err || !user)
+      return res.status('401').json({
+        error: 'User with that email does not exist!',
+      });
+
+    const token = jwt.sign(
+      { _id: user._id, iss: 'NODEAPI' },
+      process.env.JWT_SECRET
+    );
+
+    const emailData = {
+      from: '',
+      to: email,
+      subject: 'Password Reset Instructions',
+      text: `Please use the following link to reset your password: ${
+        process.env.CLIENT_URL
+      }/reset-password/${token}`,
+      html: `<p>Please use the following link to reset your password:</p> <p>${
+        process.env.CLIENT_URL
+      }/reset-password/${token}</p>`,
+    };
+
+    return user.update
+
+  });
+};
```

Comment:
In `exports.forgotPassword`, the `return user.update` line is incomplete and seems to be missing a method call. It should be `return user.updateOne()` or a similar method.

In `exports.checkUser`, there is a typo in `res.status(403).jssosn({`. It should be `res.status(403).json({`.

In `exports.forgotPassword`, the `User.findOne` callback function is not returning anything. It should return a response to the client.

In `exports.resetPassword`, the `user.save` callback function is not checking for errors properly. It should check if `err` is truthy before returning a response.

In `exports.signin`, the `User.findOne` callback function is not checking if the user is null before calling `user.authenticate`. It should check if the user exists before calling the method.